adapters/repos/db/vector/hnsw/insert.go

```diff
@@ -12,13 +12,33 @@
 package hnsw
 
 import (
+	"context"
+	"fmt"
 	"math"
 	"time"
 
 	"github.com/pkg/errors"
 	"github.com/weaviate/weaviate/adapters/repos/db/vector/hnsw/distancer"
 )
 
+func (h *hnsw) ValidateBeforeInsert(vector []float32) error {
+	if h.isEmpty() {
+		return nil
+	}
+	// check if vector length is the same as existing nodes
+	existingNodeVector, err := h.cache.get(context.Background(), h.entryPointID)
```

Comment:
smart idea to get the entry point. I was expecting you'd loop until you find the first match, but the entrypoint always has to be present, so this is a very good solution.

Reply:
thanks :)